---
title: "Enrich2 output parsing and plotting"
output: html_notebook
---

```{r}
library(ggplot2)
library(gglorenz)
library(readr)
library(stringr)
library(dplyr)
library(hrbrthemes)
library(tidyverse)
library(ggthemes)
library(ggridges)
library(ggmsa)
library(bio3d)
#library(ggbraid)
source("dms_analysis_utilities.R")
```

Import the data for plotting

```{r}
# Enrich2 score files
kir_scores_file = "../data/Enrich2/Kir2.1_sorting/tsv/Kir21_indel_exp/main_identifiers_scores.tsv"

vatd_scores_file = "../data/Enrich2/vatd/tsv/vatd_exp/main_identifiers_scores.tsv"

#Sequencing variant count files: Kir2.1
baseline_file = "/Users/bartleby/Desktop/dms_analysis/data/Kir21/baseline.tsv"

baseline <- read_delim(baseline_file, col_names = c("hgvs", "counts"), col_types = 'cd', delim='\t', skip=1)
baseline <- process_hgvs_df(baseline)

kir_variantscore_colnames <- c("hgvs", "A_SE", "A_epsilon", "A_score", "S_SE", "S_epsilon", "S_score", "SA_SE", "SA_epsilon", "SA_score")
kir_variantscore_coltypes <- c("character", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric")

vatd_variantscore_colnames <- c("hgvs", "SE_0", "epsilon_0", "score_0", "SE_016", "epsilon_016", "score_016", "SE_25", "epsilon_25", "score_25", "SE_5", "epsilon_5", "score_5")
vatd_variantscore_coltypes <- c("character", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric")

kir_scores <- read_delim(kir_scores_file, col_names = kir_variantscore_colnames, skip=3)
kir_scores <- process_hgvs_df(kir_scores)

vatd_scores <- read_delim(vatd_scores_file, col_names = vatd_variantscore_colnames, skip=3)
vatd_scores <- process_hgvs_df(vatd_scores)

vatd_scores_long <- vatd_scores %>% 
  pivot_longer(cols = starts_with("score"), names_to="VM2", names_prefix="score_", values_to = "score") %>%
  select(hgvs, pos, len, mutation_type, variants, score, VM2) %>%
  mutate(VM2 = recode(VM2, "0" = 0, "016" = 0.16, "25" = 2.5, "5" = 5), .keep="all")

oct1_scores_file ="/Users/bartleby/Desktop/dms_analysis/data/Enrich2/Oct1_full/tsv/OCT1_full_exp/main_identifiers_scores.tsv"

oct1_variantscore_colnames <- c("hgvs", "SE", "epsilon", "score")
oct1_variantscore_coltypes <- c("character", "numeric", "numeric", "numeric")

oct1_scores <- read_delim(oct1_scores_file, col_names = oct1_variantscore_colnames, skip=3)
oct1_scores <- process_hgvs_df(oct1_scores)

# Load the VatD sequence

vatd_wt_fasta = "VatD.fasta"

con=file(vatd_wt_fasta,open="r")
vatd_wt = readLines(con)[2]
close(con)

```

```{r}
oct1_tsv_scores_file ="/Users/bartleby/Desktop/dms_analysis/data/Enrich2/Oct1/tsv/OCT1_exp/main_identifiers_scores.tsv"


```

OCT1 initial plotting

```{r}

variant_df = oct1_scores %>% filter(mutation_type != "X")

order <- c("A", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L",
           "M", "N", "P", "Q", "R", "S", "T", "V", "W", "Y",
           "D_1",
           "I_1")

ggplot(data = variant_df, aes(x=pos, y=score, color=len)) + 
  geom_point() +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  labs(x = "Position") + 
  theme_classic()

ggplot(data = variant_df,
       aes(x=factor(mutation_type, level = order),
            y=score)) + 
  geom_hline(yintercept=0) +
  geom_violin(outlier.colour=NA) +
  xlab("Variant") + ylab("score") +
  labs(x = "Mutation") +
  theme_classic()

ggplot(data=variant_df, aes(pos, y=factor(variants, level = order), fill=score)) + 
  geom_tile() + scale_fill_gradient(high="blue", low="red") +
  xlab("Position") + ylab("Variant") + ggtitle("VatD score") + theme_classic()

ggplot(data = variant_df %>% 
        group_by(pos) %>% summarise(score = sum(abs(score)), pos=pos, mutation_type=mutation_type, mutation_type=mutation_type), aes(x=pos, y=score, fill=mutation_type)) +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  geom_bar(stat="identity") +
  geom_vline(xintercept = 100, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 200, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 300, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 400, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 500, linestyle = 'dashed', lty = 2) + 
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/OCT1_mean.pdf", height = 5 , width = 10)


ggplot(variant_df) +
  geom_density(aes(x = score), bw=0.01, size=1) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  facet_grid(rows = vars(mutation_type))

ggplot(variant_df, aes(x = score, y = mutation_type)) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()


#ggsave("~/Desktop/dms_analysis/Plots/OCT1_density.pdf", height = 5 , width = 6)

```

VatD plotting

```{r}

variant_df = vatd_scores_long %>% filter(mutation_type != "Z")

order <- c("A", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L",
           "M", "N", "P", "Q", "R", "S", "T", "V", "W", "Y",
           "D_1", "D_2", "D_3",
           "I_1", "I_2", "I_3", "X")

ggplot(data = variant_df %>% filter(mutation_type != "X"), aes(x=pos, y=score, color=len)) + 
  geom_point() +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  labs(x = "Position") + 
  theme_classic()

ggplot(data=variant_df, aes(pos, y=factor(variants, level = order), fill=score)) + 
  geom_tile() + scale_fill_gradient(high="blue", low="red") +
  xlab("Position") + ylab("Variant") + ggtitle("VatD score") + theme_classic()

ggplot(data = variant_df %>% 
        filter(mutation_type != "XD" & mutation_type != "S" & mutation_type != "X") %>% 
        group_by(pos) %>% summarise(score = sum(score), pos=pos, mutation_type=mutation_type, mutation_type=mutation_type), aes(x=pos, y=score, fill=mutation_type)) +
  geom_bar(stat="identity") +
  theme_classic()

# Plot DFEs - S first, then indels, last substitutions
ggplot(variant_df %>% filter(mutation_type == "S"), aes(x = score, y=factor(VM2) )) +
  geom_density_ridges(scale=2) +
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("VM2 concentration (µM)") +
  labs(colour = "Residue position") +
  theme_ridges()

ggsave("Plots/DFE_synonymous.pdf", height = 5 , width = 7)

ggplot(variant_df %>% filter(mutation_type == "I" | mutation_type == "D"), aes(x = score, y=factor(VM2) )) +
  geom_density_ridges(scale=2) +
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("VM2 concentration (µM)") +
  labs(colour = "Residue position") +
  theme_ridges()

ggsave("Plots/DFE_indel.pdf", height = 5 , width = 7)

ggplot(variant_df %>% filter(mutation_type == "M"), aes(x = score, y=factor(VM2) )) +
  geom_density_ridges(scale=2) +
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("VM2 concentration (µM)") +
  labs(colour = "Residue position") +
  theme_ridges()

ggsave("Plots/DFE_substitutions.pdf", height = 5 , width = 7)

ggplot(variant_df %>% filter(mutation_type == "M") %>% filter(VM2 == 0.16), aes(x = score, y=factor(pos) )) +
  geom_density_ridges(scale=20) +
  scale_y_discrete(expand = c(0, 0), breaks=c(2, 50, 100, 150, 200, 250)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("Position") +
  labs(colour = "Residue position") +
  ggtitle("VatD substitution Enrich2 scores (0.16 µM [VM2])") + 
  theme_ridges()

ggsave("Plots/DFE_M_016.pdf", height = 5 , width = 10)

ggplot(variant_df %>% filter(mutation_type == "I" | mutation_type == "D") %>% filter(VM2 == 0.16), aes(x = score, y=factor(pos) )) +
  geom_density_ridges(scale=20) +
  scale_y_discrete(expand = c(0, 0), breaks=c(2, 50, 100, 150, 200, 250)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("Position") +
  labs(colour = "Residue position") +
  ggtitle("VatD indel Enrich2 scores (0.16 µM [VM2])") + 
  theme_ridges()

ggsave("Plots/DFE_Indel_016.pdf", height = 5 , width = 10)

ggplot(variant_df %>% filter(mutation_type == "M") %>% filter(VM2 == 5), aes(x = score, y=factor(pos) )) +
  geom_density_ridges(scale=20) +
  scale_y_discrete(expand = c(0, 0), breaks=c(2, 50, 100, 150, 200, 250)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("Position") +
  labs(colour = "Residue position") +
  ggtitle("VatD substitution Enrich2 scores (5 µM [VM2])") + 
  theme_ridges()

ggsave("Plots/DFE_M_5.pdf", height = 5 , width = 10)

ggplot(variant_df %>% filter(mutation_type == "I" | mutation_type == "D") %>% filter(VM2 == 5), aes(x = score, y=factor(pos) )) +
  geom_density_ridges(scale=20) +
  scale_y_discrete(expand = c(0, 0), breaks=c(2, 50, 100, 150, 200, 250)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  xlab("Enrich2 score") +
  ylab("Position") +
  labs(colour = "Residue position") +
  ggtitle("VatD indel Enrich2 scores (5 µM [VM2])") + 
  theme_ridges()

ggsave("Plots/DFE_Indel_5.pdf", height = 5 , width = 10)

```

```{r}
ggplot(vatd_scores_long %>% filter(mutation_type %in% c('D', 'I'))) +
  geom_density(aes(x = score, color=factor(VM2)), bw='nrd', size=2) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  scale_color_brewer(
    type = "seq",
    palette = 'Oranges',
    direction = 1,
    aesthetics = 'color'
  ) +
  facet_grid(rows = vars(mutation_type))

#ggsave("~/Desktop/dms_analysis/Plotting/Plots/DFE.pdf", height = 5 , width = 5)

```

```{r}

physical_order <- c("G", "A", "I", "L", "V",
                    "C", "M",
                    "S", "T", 
                    "K", "R", "H",
                    "D", "E", "N", "Q",
                    "F", "W", "Y",
                    "P",
                    "D_1", "D_2", "D_3",
                    "I_1", "I_2", "I_3", "X")

vatd_wt_labels = str_split(vatd_wt, '')[[1]]

ggplot(data=vatd_scores %>% filter(pos > 0), aes(pos, y=factor(variants, level = physical_order), fill=score_0)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, low = "blue", mid = "white", high = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("VatD enrich2 scores (0 [VM2])") + 
  theme_classic(base_family="Helvetica") + 
  scale_x_continuous(breaks = seq(0, 210, by = 5)) +
    theme(text = element_text(size=25),
        axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=10),
        axis.text.x.top = element_text(size=8, family="mono", hjust=0.5)
          )

ggsave("heatmap_0VM2_2.pdf", height = 5 , width = 35)

ggplot(data=vatd_scores %>% filter(pos > 0), aes(pos, y=factor(variants, level = physical_order), fill=score_016)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, low = "blue", mid = "white", high = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("VatD enrich2 scores (0.16 µM [VM2])") + 
  theme_classic(base_family="Helvetica") + 
  scale_x_continuous(breaks = seq(0, 210, by = 5)) +
    theme(text = element_text(size=25),
        axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=10),
        axis.text.x.top = element_text(size=8, family="mono", hjust=0.5)
          )

ggsave("heatmap_016VM2_2.pdf", height = 5 , width = 35)

ggplot(data=vatd_scores %>% filter(pos > 0), aes(pos, y=factor(variants, level = physical_order), fill=score_5)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, low = "blue", mid = "white", high = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("VatD enrich2 scores (5 µM [VM2])") + 
  theme_classic(base_family="Helvetica") + 
  scale_x_continuous(breaks = seq(0, 210, by = 5)) +
    theme(text = element_text(size=25),
        axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=10),
        axis.text.x.top = element_text(size=8, family="mono", hjust=0.5)
          )

ggsave("heatmap_5VM2_2.pdf", height = 5 , width = 35)

ggplot(data=vatd_scores %>% filter(pos > 0), aes(pos, y=factor(variants, level = physical_order), fill=score_25)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, low = "blue", mid = "white", high = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("VatD enrich2 scores (2.5 µM [VM2])") + 
  theme_classic(base_family="Helvetica") + 
  scale_x_continuous(breaks = seq(0, 210, by = 5)) +
    theme(text = element_text(size=25),
        axis.text.y = element_text(size=10),
        axis.text.x = element_text(size=10),
        axis.text.x.top = element_text(size=8, family="mono", hjust=0.5)
          )

ggsave("heatmap_25VM2_2.pdf", height = 5 , width = 35)

```

```{r}



```

```{r}

fit <- lm(score_25 ~ score_0, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_0, y=score_5)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-5, 2) +
  ylim(-5, 2) +
  coord_fixed(ratio = 1) + 
  xlab("0 µM VM2") +
  ylab("5 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -3, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_0,
        vatd_scores$score_5,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_tufte(base_family="Helvetica")

fit <- lm(score_5 ~ score_016, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_016, y=score_5)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-5, 2) +
  ylim(-5, 2) +
  coord_fixed(ratio = 1) + 
  xlab("0.16 µM VM2") +
  ylab("5 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -3, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_016,
        vatd_scores$score_5,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_tufte(base_family="Helvetica")

fit <- lm(score_5 ~ score_25, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_25, y=score_5)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-5, 2) +
  ylim(-5, 2) +
  coord_fixed(ratio = 1) + 
  xlab("2.5 µM VM2") +
  ylab("5 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -3, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_25,
        vatd_scores$score_5,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_tufte(base_family="Helvetica")

fit <- lm(score_25 ~ score_016, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_25, y=score_016)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-5, 2) +
  ylim(-5, 2) +
  coord_fixed(ratio = 1) + 
  xlab("2.5 µM VM2") +
  ylab("0.16 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -3, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_25,
        vatd_scores$score_016,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_tufte(base_family="Helvetica")

fit <- lm(score_25 ~ score_0, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_25, y=score_0)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-5, 2) +
  ylim(-5, 2) +
  coord_fixed(ratio = 1) + 
  xlab("2.5 µM VM2") +
  ylab("0 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -3, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_25,
        vatd_scores$score_0,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_tufte(base_family="Helvetica")

fit <- lm(score_016 ~ score_0, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_016, y=score_0)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-3, 2) +
  ylim(-3, 2) +
  coord_fixed(ratio = 1) + 
  xlab("0.16 µM VM2") +
  ylab("0 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -2, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_016,
        vatd_scores$score_0,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_tufte(base_family="Helvetica")

```

```{r}
fit <- lm(score_016 ~ score_0, data = vatd_scores)

ggplot(data=vatd_scores, aes(x=score_016, y=score_0)) + 
  geom_point(size = 0.1) + 
  ggtitle("Enrich2 score correlations: concentrations") + 
  geom_smooth(method=lm, fullrange=TRUE, show.legend=TRUE) +
  xlim(-3, 2) +
  ylim(-3, 2) +
  coord_fixed(ratio = 1) + 
  xlab("0.16 µM VM2") +
  ylab("0 µM VM2") +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  annotate("text", x = -2, y = 1, col = "black", size = 4,
    label = paste("Pearson r = ", signif(
      cor(
        vatd_scores$score_016,
        vatd_scores$score_0,
        use = "complete.obs",
        method="pearson"),
      3))) + theme_classic()

```

```{r}
ggplot(kir_scores %>% filter(mutation_type != "S"), aes(x = S_score, y=factor(variants, level = order) )) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()

ggplot(kir_scores %>% filter(mutation_type != "S"), aes(x = A_score, y=factor(variants, level = order) )) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()


```

MSA 3D

```{r}
vat_aln = "~/Desktop/VatX/VatAD alignment.fa"
ggmsa(vat_aln, start = 221, end = 280, char_width = 0.5, seq_name = T) + geom_seqlogo() + geom_msaBar()
```

Structure mapping

```{r}
vatd_1KHR <- read.pdb("1KHR")

sse_1khr <- dssp(vatd_1KHR, exefile = "dssp", resno=TRUE, full=FALSE, verbose=FALSE)
sse_1khr_list <- data.frame(pos = strtoi(str_extract(names(sse_1khr$sse), "^[0-9]+")), SS = sse_1khr$sse)
sse_1khr_list <- sse_1khr_list %>% mutate(SS = replace(SS, SS == " ", "-"))

write.csv(sse_1khr_list, '~/Desktop/VatX/1khr_ss.csv')
```


```{r}

vatd_1KHR <- read.pdb("1KHR")

ca.chainsABC <- atom.select(vatd_1KHR, "calpha", chain = c("A", "B", "C"))
chainsABC<- atom.select(vatd_1KHR, chain = c("A", "B", "C"))

vatd_avg_scores <- vatd_scores %>% group_by(pos) %>% summarise(avg_0 = mean(score_0, na.rm=TRUE),
                                            avg_016 = mean(score_016, na.rm=TRUE), 
                                            avg_25 = mean(score_25, na.rm=TRUE),
                                            avg_5 = mean(score_5, na.rm=TRUE))

kir_avg_scores <- kir_scores %>% group_by(pos) %>% summarise(avg_A = mean(A_score, na.rm=TRUE),
                                            avg_S = mean(S_score, na.rm=TRUE), 
                                            avg_SA = mean(SA_score, na.rm=TRUE))

plot.bio3d(vatd_1KHR$atom$b[ca.chainsABC$atom], sse=vatd_1KHR, typ="l", ylab="B-factor")

x = map_scores_pdb(vatd_1KHR, vatd_avg_scores, "avg_5", selection = chainsABC)
write.pdb(x, file="1khr_5.pdb")

x = map_scores_pdb(vatd_1KHR, vatd_avg_scores, "avg_25", selection = chainsABC)
write.pdb(x, file="1khr_25.pdb")

x = map_scores_pdb(vatd_1KHR, vatd_avg_scores, "avg_016", selection = chainsABC)
write.pdb(x, file="1khr_016.pdb")

x = map_scores_pdb(vatd_1KHR, vatd_avg_scores, "avg_0", selection = chainsABC)
write.pdb(x, file="1khr_0.pdb")


kir21_pdb <- read.pdb("~/Desktop/Kir21/Kir21_alphafold.pdb")
x = map_scores_pdb(kir21_pdb, kir_avg_scores, "avg_A", selection = NULL)
write.pdb(x, file="kir21_AF_A.pdb")

x = map_scores_pdb(kir21_pdb, kir_avg_scores, "avg_S", selection = NULL)
write.pdb(x, file="kir21_AF_S.pdb")

x = map_scores_pdb(kir21_pdb, kir_avg_scores, "avg_SA", selection = NULL)
write.pdb(x, file="kir21_AF_SA.pdb")



```

Map VatA data

```{r}
vata_scores_file = "VatA_scores.tsv"

vata_variantscore_colnames <- c("hgvs", "wt_aa", "wt_codon", "pos", "mut_aa", "mut_codon", "type", "V47025", "V47050", "V47100", "VFL025", "VFL050", "VFL100", "VVM025", "VVM050", "VVM100")

vata_variantscore_coltypes <- c("character", "character", "character", "integer", "character", "character", "character", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric")

vata_scores <- read_delim(vata_scores_file, col_names = vata_variantscore_colnames, skip=3)
vata_scores <- process_hgvs_df_2(vata_scores)

#vata_scores_long <- vata_scores %>% 
#  pivot_longer(cols = starts_with("V7"), names_to="VM2", names_prefix="score_", values_to = "score") %>%
 # select(hgvs, pos, len, mutation_type, variants, score, VM2) %>%
#  mutate(VM2 = recode(VM2, "0" = 0, "016" = 0.16, "25" = 2.5, "5" = 5), .keep="all")

```

```{r}

vata_6X3C <- read.pdb("6X3C")

ca.chainsABC <- atom.select(vata_6X3C, "calpha", chain = c("A", "B", "C"))

# VatA bioassembly is chains A, E, F!
chainsABC<- atom.select(vata_6X3C, chain = c("A", "E", "F"))

vata_avg_scores <- vata_scores %>% group_by(pos) %>% summarise(avg_V47_025 = mean(V47025, na.rm=TRUE),
                                            avg_V47_050 = mean(V47050, na.rm=TRUE), 
                                            avg_V47_100 = mean(V47100, na.rm=TRUE),
                                            avg_VFL_025 = mean(VFL025, na.rm=TRUE),
                                            avg_VFL_050 = mean(VFL050, na.rm=TRUE), 
                                            avg_VFL_100 = mean(VFL100, na.rm=TRUE),
                                            avg_VVM_025 = mean(VVM025, na.rm=TRUE),
                                            avg_VVM_050 = mean(VVM050, na.rm=TRUE), 
                                            avg_VVM_100 = mean(VVM100, na.rm=TRUE))

vata_var_scores <- vata_scores %>% group_by(pos) %>% summarise(var_V47_025 = var(V47025, na.rm=TRUE),
                                            var_V47_050 = var(V47050, na.rm=TRUE), 
                                            var_V47_100 = var(V47100, na.rm=TRUE),
                                            var_VFL_025 = var(VFL025, na.rm=TRUE),
                                            var_VFL_050 = var(VFL050, na.rm=TRUE), 
                                            var_VFL_100 = var(VFL100, na.rm=TRUE),
                                            var_VVM_025 = var(VVM025, na.rm=TRUE),
                                            var_VVM_050 = var(VVM050, na.rm=TRUE), 
                                            var_VVM_100 = var(VVM100, na.rm=TRUE))

plot.bio3d(vata_6X3C$atom$b[ca.chainsABC$atom], sse=vata_6X3C, typ="l", ylab="B-factor")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_V47_025", selection = chainsABC)
write.pdb(x, file="6X3C_v47_avg_025.pdb")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_V47_050", selection = chainsABC)
write.pdb(x, file="6X3C_v47_avg_050.pdb")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_V47_100", selection = chainsABC)
write.pdb(x, file="6X3C_v47_avg_100.pdb")


x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_VFL_025", selection = chainsABC)
write.pdb(x, file="6X3C_vfl_avg_025.pdb")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_VFL_050", selection = chainsABC)
write.pdb(x, file="6X3C_vfl_avg_050.pdb")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_VFL_100", selection = chainsABC)
write.pdb(x, file="6X3C_vfl_avg_100.pdb")


x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_VVM_025", selection = chainsABC)
write.pdb(x, file="6X3C_vvm_avg_025.pdb")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_VVM_050", selection = chainsABC)
write.pdb(x, file="6X3C_vvm_avg_050.pdb")

x = map_scores_pdb(vata_6X3C, vata_avg_scores, "avg_VVM_100", selection = chainsABC)
write.pdb(x, file="6X3C_vvm_avg_100.pdb")

# Write variance PDBs
x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_V47_025", selection = chainsABC)
write.pdb(x, file="6X3C_v47_var_025.pdb")

x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_V47_050", selection = chainsABC)
write.pdb(x, file="6X3C_v47_var_050.pdb")

x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_V47_100", selection = chainsABC)
write.pdb(x, file="6X3C_v47_var_100.pdb")


x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_VFL_025", selection = chainsABC)
write.pdb(x, file="6X3C_vfl_var_025.pdb")

x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_VFL_050", selection = chainsABC)
write.pdb(x, file="6X3C_vfl_var_050.pdb")

x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_VFL_100", selection = chainsABC)
write.pdb(x, file="6X3C_vfl_var_100.pdb")


x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_VVM_025", selection = chainsABC)
write.pdb(x, file="6X3C_vvm_var_025.pdb")

x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_VVM_050", selection = chainsABC)
write.pdb(x, file="6X3C_vvm_var_050.pdb")

x = map_scores_pdb(vata_6X3C, vata_var_scores, "var_VVM_100", selection = chainsABC)
write.pdb(x, file="6X3C_vvm_var_100.pdb")

```

```{r}

variant_df = kir_scores %>% filter(mutation_type != "X")

order <- c("A", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L",
           "M", "N", "P", "Q", "R", "S", "T", "V", "W", "Y",
           "D_1", "D_2", "D_3",
           "I_1", "I_2", "I_3")

ggplot(data = variant_df, aes(x=pos, y=A_score, color=len)) + 
  geom_point() +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  labs(x = "Position") + 
  theme_classic()

ggplot(data = variant_df,
       aes(x=factor(mutation_type, level = order),
            y=A_score)) + 
  geom_hline(yintercept=0) +
  geom_violin(outlier.colour=NA) +
  xlab("Variant") + ylab("score") +
  labs(x = "Mutation") +
  theme_classic()

ggplot(data=variant_df, aes(pos, y=factor(variants, level = order), fill=A_score)) + 
  geom_tile() + scale_fill_gradient(high="blue", low="red") +
  xlab("Position") + ylab("Variant") + ggtitle("Kir2.1 Abundance score") + theme_classic()

ggplot(data=variant_df, aes(pos, y=factor(variants, level = order), fill=S_score)) + 
  geom_tile() + scale_fill_gradient(high="blue", low="red") +
  xlab("Position") + ylab("Variant") + ggtitle("Kir2.1 Surface score") + theme_classic()

ggplot(data = variant_df %>% 
        group_by(pos) %>% summarise(score = sum(abs(A_score)), pos=pos, mutation_type=mutation_type, mutation_type=mutation_type), aes(x=pos, y=score, fill=mutation_type)) +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  geom_bar(stat="identity") +
  geom_vline(xintercept = 100, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 200, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 300, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 400, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 500, linestyle = 'dashed', lty = 2) + 
  theme_classic()

ggplot(variant_df) +
  geom_density(aes(x = A_score), bw=0.01, size=1) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  facet_grid(rows = vars(mutation_type))

ggplot(variant_df, aes(x = A_score, y = mutation_type)) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()

```

Plot OCT1 full results

```{r}
oct1_scores_file ="/Users/bartleby/Desktop/dms_analysis/data/Enrich2/Oct1_full/tsv/OCT1_full_exp/main_identifiers_scores.tsv"
oct1_scores_shared_file ="/Users/bartleby/Desktop/dms_analysis/data/Enrich2/Oct1_full/tsv/OCT1_full_exp/main_identifiers_scores_shared.tsv"

oct1_variantscore_colnames <- c("hgvs", "SM73_0_SE", "SM73_0_epsilon", "SM73_0_score", "SM73_1_SE", "SM73_1_epsilon", "SM73_1_score", "GFP_SE", "GFP_epsilon", "GFP_score")
oct1_variantscore_coltypes <- c("character", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric", "numeric")

oct1_shared_colnames <- c("hgvs", "SM73_0_SE_R1", "SM73_0_score_R1",
                                "SM73_0_SE_R2", "SM73_0_score_R2",
                                "SM73_0_SE_R3", "SM73_0_score_R3",
                                "SM73_1_SE_R1", "SM73_1_score_R1",
                                "SM73_1_SE_R2", "SM73_1_score_R2",
                                "SM73_1_SE_R3", "SM73_1_score_R3",
                                "GFP_SE_R1", "GFP_score_R1",
                                "GFP_SE_R2", "GFP_score_R2",
                                "GFP_SE_R3", "GFP_score_R3")
oct1_shared_coltypes <- c("character", "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric",
                                "numeric", "numeric")

oct1_scores <- read_delim(oct1_scores_file, col_names = oct1_variantscore_colnames, skip=3)
oct1_scores <- process_hgvs_df(oct1_scores)

oct1_scores_shared <- read_delim(oct1_scores_shared_file, col_names = oct1_shared_colnames, skip=5)

oct1_scores_shared <- oct1_scores_shared %>%  rowwise() %>%
                                mutate(SM73_0_score = mean(c_across(starts_with("SM73_0_score")), na.rm=TRUE),
                                SM73_1_score = mean(c_across(starts_with("SM73_1_score")), na.rm=TRUE),
                                GFP_score = mean(c_across(starts_with("GFP_score")), na.rm=TRUE))

oct1_scores_shared <- process_hgvs_df(oct1_scores_shared %>% select(c(hgvs, SM73_0_score, SM73_1_score, GFP_score)))
oct1_scores_shared$is.wt = oct1_scores_shared$mutation_type == "S"
oct1_scores <- oct1_scores_shared

```

```{r}


ggplot(oct1_scores %>% filter(mutation_type != "X")) +
  geom_density(aes(x = SM73_0_score), size=2) +
  xlab("Enrich2 score (0 µM [SM73])") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  scale_color_brewer(
    type = "seq",
    palette = 'Oranges',
    direction = 1,
    aesthetics = 'color'
  ) +
  facet_grid(rows = vars(mutation_type))

ggplot(oct1_scores %>% filter(mutation_type != "X")) +
  geom_density(aes(x = SM73_1_score), size=2) +
  xlab("Enrich2 score (1 µM [SM73])") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  scale_color_brewer(
    type = "seq",
    palette = 'Oranges',
    direction = 1,
    aesthetics = 'color'
  ) +
  facet_grid(rows = vars(mutation_type))

ggplot(oct1_scores %>% filter(mutation_type != "X")) +
  geom_density(aes(x = GFP_score), size=2) +
  xlab("Enrich2 score (GFP)") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  scale_color_brewer(
    type = "seq",
    palette = 'Oranges',
    direction = 1,
    aesthetics = 'color'
  ) +
  facet_grid(rows = vars(mutation_type))

```

```{r}

physical_order <- c("G", "A", "I", "L", "V",
                    "C", "M",
                    "S", "T", 
                    "K", "R", "H",
                    "D", "E", "N", "Q",
                    "F", "W", "Y",
                    "P",
                    "D_1"
                    )

ggplot(data=oct1_scores, aes(pos, y=factor(variants, level = physical_order), fill=SM73_0_score)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, high = "blue", mid = "white", low = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("OCT1 Enrich scores (0 µM [SM73])") + 
  theme_classic(base_family="Helvetica") + 
  theme(text = element_text(size=25),
        axis.text.y = element_text(size=5)
          )
ggsave("Plots/OCT1_heatmap_0SM73.pdf", height = 9 , width = 35)


ggplot(data=oct1_scores, aes(pos, y=factor(variants, level = physical_order), fill=SM73_1_score)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, high = "blue", mid = "white", low = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("OCT1 Enrich scores (1 µM [SM73])") + 
  theme_classic(base_family="Helvetica") + 
  theme(text = element_text(size=25),
        axis.text.y = element_text(size=5)
          )
ggsave("Plots/OCT1_heatmap_1SM73.pdf", height = 9 , width = 35)


ggplot(data=oct1_scores, aes(pos, y=factor(variants, level = physical_order), fill=GFP_score)) + 
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 0.1) +
  scale_fill_gradient2(midpoint = 0, low = "blue", mid = "white", high = "red") + 
  coord_fixed() + 
  xlab("Position") + ylab("Variant") + ggtitle("OCT1 Enrich scores (Abundance)") + 
  theme_classic(base_family="Helvetica") + 
  theme(text = element_text(size=25),
        axis.text.y = element_text(size=5)
          )
ggsave("Plots/OCT1_heatmap_GFP.pdf", height = 9 , width = 35)

```

```{r}

variant_df = oct1_scores %>% filter(mutation_type != "X")

physical_order <- c("G", "A", "I", "L", "V",
                    "C", "M",
                    "S", "T", 
                    "K", "R", "H",
                    "D", "E", "N", "Q",
                    "F", "W", "Y",
                    "P",
                    "D_1")

ggplot(data = variant_df, aes(x=pos, y=GFP_score)) + 
  geom_point() +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  labs(x = "Position", y = "Enrich score  (Abundance)") + 
  theme_classic()

ggplot(data = variant_df,
       aes(x=factor(mutation_type, level = physical_order),
            y=GFP_score)) + 
  geom_hline(yintercept=0) +
  geom_violin(outlier.colour=NA) +
  xlab("Variant") + ylab("Enrich score  (Abundance)") +
  labs(x = "Mutation") +
  theme_classic()

ggplot(data = variant_df %>% 
        group_by(pos) %>% summarise(score = mean(GFP_score, na.rm=TRUE), pos = first(pos), mutation_type=first(mutation_type)), aes(x=pos, y=score, fill=mutation_type)) +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  geom_bar(stat="identity") +
  geom_vline(xintercept = 100, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 200, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 300, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 400, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 500, linestyle = 'dashed', lty = 2) + 
  theme_classic()

ggplot(variant_df) +
  geom_density(aes(x = GFP_score), bw=0.01, size=1) +
  xlab("Enrich2 score (Abundance)") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  facet_grid(rows = vars(mutation_type))

ggplot(variant_df, aes(x = GFP_score, y = mutation_type)) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score (Abundance)") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()

```

```{r}
variant_df = oct1_scores %>% filter(mutation_type != "X")

physical_order <- c("G", "A", "I", "L", "V",
                    "C", "M",
                    "S", "T", 
                    "K", "R", "H",
                    "D", "E", "N", "Q",
                    "F", "W", "Y",
                    "P",
                    "D_1")

ggplot(data = variant_df, aes(x=pos, y=SM73_0_score)) + 
  geom_point() +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  labs(x = "Position", y = "Enrich score  (0 µM [SM73])") + 
  theme_classic()

ggplot(data = variant_df,
       aes(x=factor(mutation_type, level = physical_order),
            y=SM73_0_score)) + 
  geom_hline(yintercept=0) +
  geom_violin(outlier.colour=NA) +
  xlab("Variant") + ylab("score") +
  labs(x = "Mutation") +
  theme_classic()

ggplot(data = variant_df %>% 
        group_by(pos) %>% summarise(score = mean(SM73_0_score, na.rm=TRUE), pos = first(pos), mutation_type=first(mutation_type)), aes(x=pos, y=score, fill=mutation_type)) +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  geom_bar(stat="identity") +
  geom_vline(xintercept = 100, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 200, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 300, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 400, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 500, linestyle = 'dashed', lty = 2) + 
  theme_classic()

ggplot(variant_df) +
  geom_density(aes(x = SM73_0_score), bw=0.01, size=1) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  facet_grid(rows = vars(mutation_type))

ggplot(variant_df, aes(x = SM73_0_score, y = mutation_type)) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()

```

```{r}
variant_df = oct1_scores %>% filter(mutation_type != "X")

physical_order <- c("G", "A", "I", "L", "V",
                    "C", "M",
                    "S", "T", 
                    "K", "R", "H",
                    "D", "E", "N", "Q",
                    "F", "W", "Y",
                    "P",
                    "D_1")

ggplot(data = variant_df, aes(x=pos, y=SM73_1_score)) + 
  geom_point() +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  labs(x = "Position") + 
  theme_classic()

ggplot(data = variant_df,
       aes(x=factor(mutation_type, level = physical_order),
            y=SM73_1_score)) + 
  geom_hline(yintercept=0) +
  geom_violin(outlier.colour=NA) +
  xlab("Variant") + ylab("score") +
  labs(x = "Mutation") +
  theme_classic()

ggplot(data = variant_df %>% 
        group_by(pos) %>% summarise(score = mean(SM73_1_score, na.rm=TRUE), pos = first(pos), mutation_type=first(mutation_type)), aes(x=pos, y=score, fill=mutation_type)) +
  facet_grid(rows = vars(mutation_type), scales = "free") +
  geom_bar(stat="identity") +
  geom_vline(xintercept = 100, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 200, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 300, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 400, linestyle = 'dashed', lty = 2) + 
  geom_vline(xintercept = 500, linestyle = 'dashed', lty = 2) + 
  theme_classic()

ggplot(variant_df) +
  geom_density(aes(x = SM73_1_score), bw=0.01, size=1) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  facet_grid(rows = vars(mutation_type))

ggplot(variant_df, aes(x = SM73_1_score, y = mutation_type)) +
  geom_density_ridges(scale=1) +
  geom_point(aes(color=pos), 
    position = position_jitter(height=0.05, width = 0), size=0.25) + 
  scale_y_discrete(expand = c(0, 0)) +     # will generally have to set the `expand` option
  scale_x_continuous(expand = c(0, 0)) +   # for both axes to remove unneeded padding
  coord_cartesian(clip = "off") + # to avoid clipping of the very top of the top ridgeline
  scale_color_gradient(high = "#49006a", low = "#fff7f3", aesthetics = "color") +
  xlab("Enrich2 score") +
  ylab("Mutation type") +
  labs(colour = "Residue position") +
  theme_ridges()
```
OCT1 heatmap
```{r}

order <- c('D_1','H', 'K', 'R', 'D', 'E',
          'C','M','N','Q','S','T','A','I','L','V','W','F',
          'Y','G','P')

oct1_wt = "MPTVDDILEQVGESGWFQKQAFLILCLLSAAFAPICVGIVFLGFTPDHHCQSPGVAELSQRCGWSPAEELNYTVPGLGPAGEAFLGQCRRYEVDWNQSALSCVDPLASLATNRSHLPLGPCQDGWVYDTPGSSIVTEFNLVCADSWKLDLFQSCLNAGFLFGSLGVGYFADRFGRKLCLLGTVLVNAVSGVLMAFSPNYMSMLLFRLLQGLVSKGNWMAGYTLITEFVGSGSRRTVAIMYQMAFTVGLVALTGLAYALPHWRWLQLAVSLPTFLFLLYYWCVPESPRWLLSQKRNTEAIKIMDHIAQKNGKLPPADLKMLSLEEDVTEKLSPSFADLFRTPRLRKRTFILMYLWFTDSVLYQGLILHMGATSGNLYLDFLYSALVEIPGAFIALITIDRVGRIYPMAMSNLLAGAACLVMIFISPDLHWLNIIIMCVGRMGITIAIQMICLVNAELYPTFVRNLGVMVCSSLCDIGGIITPFIVFRLREVWQALPLILFAVLGLLAAGVTLLLPETKGVALPETMKDAENLGRKAKPKENTIYLKVQTSEPSGT"

oct_wt_1 = str_split(substr(oct1_wt, 1, 100), '')[[1]]
oct_wt_2 = str_split(substr(oct1_wt, 101, 200), '')[[1]]
oct_wt_3 = str_split(substr(oct1_wt, 201, 300), '')[[1]]
oct_wt_4 = str_split(substr(oct1_wt, 301, 400), '')[[1]]
oct_wt_5 = str_split(substr(oct1_wt, 401, 500), '')[[1]]
oct_wt_6 = str_split(substr(oct1_wt, 501, 550), '')[[1]]

#surface plot
SM73_1_1 = ggplot(data = oct1_scores[oct1_scores$pos %in% c(1:100),], 
                      aes(x = pos, y = factor(variants, level = order), fill = SM73_1_score)) +
  geom_tile(aes(color = as.factor(is.wt)), size = 0.2) +
  scale_fill_continuous_divergingx(palette = 'RdBu', mid = 0,l1=0.2, l3 = 0.2, p1=0.9, p3 = .4, p4 = 0.9,rev=TRUE, na.value = 'lightyellow',limits=c(-3,3)) + 
  scale_color_manual(values = c(NA,'green')) +
  scale_x_continuous(breaks = seq(0, 550, by = 5),
                     expand = c(0,0),
                     sec.axis = sec_axis(
                      trans = ~.,
                      name = "Sequence",
                      breaks = seq(1, 100),
                      labels = oct_wt_1,
                      guide = derive()
                    )) +
  coord_fixed(ratio = 1) +
  theme(
    panel.background = element_rect(fill = "white",
                                    colour = "white",
                                    size = 0.5, linetype = "solid"),
    panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                    colour = "lightgray"), 
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.text = element_text(size = 5),
    axis.text.x.top = element_text(angle = 0, hjust = 0.5)
  ) +
  labs(y = "Mutation", x = "Position")

SM73_1_2 = ggplot(data = oct1_scores[oct1_scores$pos %in% c(101:200),], 
                      aes(x = pos, y = factor(variants, level = order), fill = SM73_1_score)) +
  geom_tile(aes(color = as.factor(is.wt)), size = 0.2) +
  scale_fill_continuous_divergingx(palette = 'RdBu', mid = 0,l1=0.2, l3 = 0.2, p1=0.9, p3 = .4, p4 = 0.9,rev=TRUE, na.value = 'lightyellow',limits=c(-5,5)) + 
  scale_color_manual(values = c(NA,'green')) +
  scale_x_continuous(breaks = seq(0, 550, by = 5),
                     expand = c(0,0),
                     sec.axis = sec_axis(
                      trans = ~.,
                      name = "Sequence",
                      breaks = seq(101, 200),
                      labels = oct_wt_2,
                      guide = derive()
                    )) +
  coord_fixed(ratio = 1) +
  theme(
    panel.background = element_rect(fill = "white",
                                    colour = "white",
                                    size = 0.5, linetype = "solid"),
    panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                    colour = "lightgray"), 
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.text = element_text(size = 5),
    axis.text.x.top = element_text(angle = 0, hjust = 0.5)
  ) +
  labs(y = "Mutation", x = "Position")

SM73_1_3 = ggplot(data = oct1_scores[oct1_scores$pos %in% c(201:300),], 
                      aes(x = pos, y = factor(variants, level = order), fill = SM73_1_score)) +
  geom_tile(aes(color = as.factor(is.wt)), size = 0.2) +
  scale_fill_continuous_divergingx(palette = 'RdBu', mid = 0,l1=0.2, l3 = 0.2, p1=0.9, p3 = .4, p4 = 0.9,rev=TRUE, na.value = 'lightyellow',limits=c(-3,3)) + 
  scale_color_manual(values = c(NA,'green')) +
  scale_x_continuous(breaks = seq(0, 550, by = 5),
                     expand = c(0,0),
                     sec.axis = sec_axis(
                      trans = ~.,
                      name = "Sequence",
                      breaks = seq(201, 300),
                      labels = oct_wt_3,
                      guide = derive()
                    )) +
  coord_fixed(ratio = 1) +
  theme(
    panel.background = element_rect(fill = "white",
                                    colour = "white",
                                    size = 0.5, linetype = "solid"),
    panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                    colour = "lightgray"), 
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.text = element_text(size = 5),
    axis.text.x.top = element_text(angle = 0, hjust = 0.5)
  ) +
  labs(y = "Mutation", x = "Position")

SM73_1_4 = ggplot(data = oct1_scores[oct1_scores$pos %in% c(301:400),], 
                      aes(x = pos, y = factor(variants, level = order), fill = SM73_1_score)) +
  geom_tile(aes(color = as.factor(is.wt)), size = 0.2) +
  scale_fill_continuous_divergingx(palette = 'RdBu', mid = 0,l1=0.2, l3 = 0.2, p1=0.9, p3 = .4, p4 = 0.9,rev=TRUE, na.value = 'lightyellow',limits=c(-3,3)) + 
  scale_color_manual(values = c(NA,'green')) +
  scale_x_continuous(breaks = seq(0, 550, by = 5),
                     expand = c(0,0),
                     sec.axis = sec_axis(
                      trans = ~.,
                      name = "Sequence",
                      breaks = seq(301, 400),
                      labels = oct_wt_4,
                      guide = derive()
                    )) +
  coord_fixed(ratio = 1) +
  theme(
    panel.background = element_rect(fill = "white",
                                    colour = "white",
                                    size = 0.5, linetype = "solid"),
    panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                    colour = "lightgray"), 
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.text = element_text(size = 5),
    axis.text.x.top = element_text(angle = 0, hjust = 0.5)
  ) +
  labs(y = "Mutation", x = "Position")


SM73_1_5 = ggplot(data = oct1_scores[oct1_scores$pos %in% c(401:500),], 
                      aes(x = pos, y = factor(variants, level = order), fill = SM73_1_score)) +
  geom_tile(aes(color = as.factor(is.wt)), size = 0.2) +
  scale_fill_continuous_divergingx(palette = 'RdBu', mid = 0,l1=0.2, l3 = 0.2, p1=0.9, p3 = .4, p4 = 0.9,rev=TRUE, na.value = 'lightyellow',limits=c(-3,3)) + 
  scale_color_manual(values = c(NA,'green')) +
  scale_x_continuous(breaks = seq(0, 550, by = 5),
                     expand = c(0,0),
                     sec.axis = sec_axis(
                      trans = ~.,
                      name = "Sequence",
                      breaks = seq(401, 500),
                      labels = oct_wt_5,
                      guide = derive()
                    )) +
  coord_fixed(ratio = 1) +
  theme(
    panel.background = element_rect(fill = "white",
                                    colour = "white",
                                    size = 0.5, linetype = "solid"),
    panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                    colour = "lightgray"), 
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.text = element_text(size = 5),
    axis.text.x.top = element_text(angle = 0, hjust = 0.5)
  ) +
  labs(y = "Mutation", x = "Position")

SM73_1_6 = ggplot(data = oct1_scores[oct1_scores$pos %in% c(501:550),], 
                      aes(x = pos, y = factor(variants, level = order), fill = SM73_1_score)) +
  geom_tile(aes(color = as.factor(is.wt)), size = 0.2) +
  scale_fill_continuous_divergingx(palette = 'RdBu', mid = 0,l1=0.2, l3 = 0.2, p1=0.9, p3 = .4, p4 = 0.9,rev=TRUE, na.value = 'lightyellow',limits=c(-3,3)) + 
  scale_color_manual(values = c(NA,'green')) +
  scale_x_continuous(breaks = seq(0, 550, by = 5),
                     expand = c(0,0),
                     sec.axis = sec_axis(
                      trans = ~.,
                      name = "Sequence",
                      breaks = seq(501, 550),
                      labels = oct_wt_6,
                      guide = derive()
                    )) +
  coord_fixed(ratio = 1) +
  theme(
    panel.background = element_rect(fill = "white",
                                    colour = "white",
                                    size = 0.5, linetype = "solid"),
    panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                    colour = "lightgray"), 
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.text = element_text(size = 5),
    axis.text.x.top = element_text(angle = 0, hjust = 0.5)
  ) +
  labs(y = "Mutation", x = "Position")


SM73_DMS = ggarrange(SM73_1_1, SM73_1_2, SM73_1_3, SM73_1_4, SM73_1_5, SM73_1_6,
                        nrow = 6, ncol = 1)
SM73_DMS
annotate_figure(SM73_DMS, fig.lab = '1 µM [SM73]')

ggsave("~/Desktop/dms_analysis/Plotting/Plots/OCT1_SM731_heatmap.pdf", height = 11, width = 8.5)
ggsave("~/Desktop/dms_analysis/Plotting/Plots/OCT1_SM731_heatmap.png", height = 11, width = 8.5)

```
```


GMM?

```{r}

X <- vatd_scores[c('score_0', 'score_5')]
X <- as.data.frame(X, na.rm=TRUE)
BIC <- mclustBIC(X)
plot(BIC)

summary(BIC)

mod1 <- Mclust(X, x = BIC)
summary(mod1, parameters = TRUE)

plot(mod1, what = "classification")
plot(mod1, what = "uncertainty")
ICL <- mclustICL(X)
summary(ICL)


mod1 <- Mclust(X, x = BIC, G=4)
summary(mod1, parameters = TRUE)
plot(mod1, what = "classification")
plot(mod1, what = "uncertainty")

```

Map the OCT1 results to the Alphafold model
```{r}

# Single-chain model.
OCT1_AF <- read.pdb("Structures/AF-O15245-F1-model_v2.pdb")

oct1_avg_scores <- oct1_scores %>% group_by(pos) %>% summarise(avg_0_SM73 = mean(SM73_0_score, na.rm=TRUE),
                                            avg_1_SM73 = mean(SM73_1_score, na.rm=TRUE), 
                                            avg_A = mean(GFP_score, na.rm=TRUE))

plot.bio3d(OCT1_AF$atom$b, sse=OCT1_AF, typ="l", ylab="B-factor")

x = map_scores_pdb(OCT1_AF, oct1_avg_scores, "avg_0_SM73")
write.pdb(x, file="Structures/oct1_AF_avg_0_2.pdb")

x = map_scores_pdb(OCT1_AF, oct1_avg_scores, "avg_1_SM73")
write.pdb(x, file="Structures/oct1_AF_avg_1_2.pdb")

x = map_scores_pdb(OCT1_AF, oct1_avg_scores, "avg_A")
write.pdb(x, file="Structures/oct1_AF_avg_A_2.pdb")


```

```{r}

variant_df = kir_scores %>% filter(mutation_type != "X")

order <- c("A", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L",
           "M", "N", "P", "Q", "R", "S", "T", "V", "W", "Y",
           "D_1", "D_2", "D_3",
           "I_1", "I_2", "I_3")

ggplot(data = variant_df, aes(x=A_score, y=S_score), na.rm=TRUE) + 
  geom_density_2d(data = kir_scores %>% filter(mutation_type == "M"), color='blue') +
    geom_point(data = kir_scores %>% filter(mutation_type == "M"), color='blue', alpha=0.2) +
  labs(x = "Abundance", y = "Surface") + 
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/Kir21_contours", height = 5 , width = 10)
```

```{r}
X <- variant_df[c('S_score', 'A_score')]
X <- as.data.frame(X, na.rm=TRUE)
BIC <- mclustBIC(X)
plot(BIC)

summary(BIC)

mod1 <- Mclust(X, x = BIC)
summary(mod1, parameters = TRUE)

plot(mod1, what = "classification")
plot(mod1, what = "uncertainty")
ICL <- mclustICL(X)
summary(ICL)

```


Kir2.1 plotting

```{r}




```




Let's examine epistasis in the indel substitutions.

```{r}
indel_scores <- indel_scores %>% ungroup()

# This needs to be _lead_, since the position of deletions is set to be the
# _first_ deleted codon, not the last.

indel_scores <- indel_scores %>% mutate(D_1, 
                          epi_D_2 = (D_1 + lead(D_1, order_by=pos)),
                          epi_D_3 = (D_1 + lead(D_1, order_by=pos) + lead(D_1, n=2, order_by=pos)),
                          epi_D_3_r21 = (D_1 + lead(D_2, order_by=pos)),
                          epi_D_3_l21 = (D_2 + lead(D_1, order_by=pos, n=2)))

indel_scores <- indel_scores %>% mutate(dD2 = (D_2 - epi_D_2),
                                       dD3 = (D_3 - epi_D_3),
                                       dD3_r21 = (D_3 - epi_D_3_r21),
                                       dD3_l21 = (D_3 - epi_D_3_l21))

```

```{r}
kir21_3spi_pdb <- read.pdb("structures/3spi.pdb")

sse_3spi <- dssp(kir21_3spi_pdb, exefile = "dssp", resno=TRUE, full=FALSE, verbose=FALSE)

sse_list <- data.frame(pos = strtoi(str_extract(names(sse_3spi$sse), "^[0-9]+")), SS = sse_3spi$sse)

indel_scores <- left_join(indel_scores, data.frame(sse_list), by='pos')
```

Condition indels on secondary structure
```{r}
indel_scores_long <- indel_scores %>% filter(!is.na(SS)) %>%
                            pivot_longer(cols = contains("_"),
                                        names_sep=c("_"),
                                        names_to=c("type", "length"),
                                        values_to = "score")

  indel_scores_long <- indel_scores_long %>% mutate(SS = replace(SS, SS == " ", "C")) %>%
                        mutate(SS = replace(SS, SS == "G", "H")) %>%
                        mutate(SS = replace(SS, SS == "I", "H")) %>%
                        mutate(SS = replace(SS, SS == "E", "B"))



ggplot(
  data = indel_scores_long %>% filter(!is.na(SS)) %>%
                              mutate(SS = replace(SS, SS == " ", "C")) %>%
                              mutate(SS = replace(SS, SS == "G", "H")) %>%
                              mutate(SS = replace(SS, SS == "I", "H")) %>%
                              mutate(SS = replace(SS, SS == "E", "B"))
  ) +
  facet_grid(rows = vars(SS)) +
  geom_density(aes(x = score, color = type), size=1) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  scale_color_brewer(
    type = "qual",
    palette = 'Dark2',
    direction = -1,
  ) + 
  labs(color = "Mutation type")

kir_scores <- left_join(kir_scores, data.frame(sse_list), by='pos') %>% 
                        mutate(SS = replace(SS, SS == " ", "C")) %>%
                        mutate(SS = replace(SS, SS == "G", "H")) %>%
                        mutate(SS = replace(SS, SS == "I", "H")) %>%
                        mutate(SS = replace(SS, SS == "E", "B"))

ggplot(
  data = kir_scores %>% filter(!is.na(SS)) %>% filter(mutation_type != "S" & mutation_type != "X")
  ) +
  facet_grid(rows = vars(SS), cols = vars(len)) +
  geom_density(aes(x = S_score, color = mutation_type), size=1) +
  xlab("Enrich2 score") + 
  ylab("Density") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme_classic(base_family = "Helvetica") + 
  scale_color_brewer(
    type = "qual",
    palette = 'Dark2',
    direction = -1,
  ) + 
  labs(color = "Mutation type")

```


```{r}
ggplot(data = indel_scores, aes(x=pos, y=D_1), na.rm=TRUE) + 
    geom_hline(yintercept=0) +
    geom_tile(
    aes(x=Kir2_1_FLAG_Resno_mus, y=y,height=height, fill = factor(SS, exclude = NA)), 
    ymin = 10, ymax = 15, alpha = 0.2,
    data = subset(ss, !is.na(SS))
  ) +  scale_fill_manual(values = c("H" = 'red',
                                    "G" = 'orange',
                                    "I" = 'white',
                                    "B" = 'white',
                                    "E" = 'blue',
                                    " " = 'white',
                                    "S" = 'white',
                                    "T" = 'white',
                                    "C" = 'white')) +
  geom_line(aes(x=pos, y=dD2), color='red') +
  geom_line(aes(x=pos, y=dD3), color='blue') +
  labs(x = "Position", y = "Epistasis (Actual - single)") + 
  guides(fill="none") +
  theme_classic()

#ggsave("~/Desktop/dms_analysis/Plotting/Plots/Kir21_S_single_epistastis.png", height = 5 , width = 10)

ggplot(data = indel_scores, aes(x=pos, y=D_2), na.rm=TRUE) + 
    geom_hline(yintercept=0) +
    geom_tile(
    aes(x=Kir2_1_FLAG_Resno_mus, y=y,height=height, fill = factor(SS, exclude = NA)), 
    ymin = 10, ymax = 15, alpha = 0.2,
    data = subset(ss, !is.na(SS))
  ) +  scale_fill_manual(values = c("H" = 'red',
                                    "G" = 'orange',
                                    "I" = 'white',
                                    "B" = 'white',
                                    "E" = 'blue',
                                    " " = 'white',
                                    "S" = 'white',
                                    "T" = 'white',
                                    "C" = 'white')) +
  geom_line(aes(x=pos, y=dD3), color='blue') +
  geom_line(aes(x=pos, y=dD3_r21), color='darksalmon') +
  geom_line(aes(x=pos, y=dD3_l21), color='darkgoldenrod') +
  labs(x = "Position", y = "Epistasis (Actual - single)") + 
  guides(fill="none") +
  theme_classic()




```



```{r}

ggplot(data = indel_scores) +
  scale_x_continuous(breaks=c(1, 50,100, 150, 200, 250, 300, 350, 400)) +
  geom_line(aes(pos, epi_D_2), linetype=2) +
  geom_hline(yintercept=0) +
  geom_line(aes(pos, D_2), linetype=1) +
  geom_braid(aes(pos, ymin = epi_D_2, ymax = D_2, fill = epi_D_2 < D_2), alpha = 0.5) +
  guides(fill="none") +
  xlab("Position") + ylab("Surface score") +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/Kir21_S_epi2_braid.png", height = 5 , width = 10)

ggplot(data = indel_scores) +
  scale_x_continuous(breaks=c(1, 50,100, 150, 200, 250, 300, 350, 400)) +
  geom_line(aes(pos, epi_D_3), linetype=2) +
  geom_hline(yintercept=0) +
  geom_line(aes(pos, D_3), linetype=1) +
  geom_braid(aes(pos, ymin = epi_D_3, ymax = D_3, fill = epi_D_3 < D_3), alpha = 0.5) +
  guides(fill="none") +
  xlab("Position") + ylab("Surface score") +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/Kir21_S_epi3_braid.png", height = 5 , width = 10)


```

VatD indel epistasis
```{r}

vatd_indel_scores_25 <- vatd_scores %>% filter(mutation_type == "D" | mutation_type == "I") %>% arrange(pos) %>% group_by(pos) %>%
    pivot_wider(id_cols = c(pos), names_from = variants,
                values_from = score_25,
                values_fill = 0,
                names_expand = TRUE)

vatd_indel_scores_25 <- vatd_indel_scores_25 %>% ungroup()

# This needs to be _lead_, since the position of deletions is set to be the
# _first_ deleted codon, not the last.

vatd_indel_scores_25 <- vatd_indel_scores_25 %>% mutate(D_1, 
                          epi_D_2 = (D_1 + lead(D_1, order_by=pos)),
                          epi_D_3 = (D_1 + lead(D_1, order_by=pos) + lead(D_1, n=2, order_by=pos)),
                          epi_D_3_r21 = (D_1 + lead(D_2, order_by=pos)),
                          epi_D_3_l21 = (D_2 + lead(D_1, order_by=pos, n=2)))

vatd_indel_scores_25 <- vatd_indel_scores_25 %>% mutate(dD2 = (D_2 - epi_D_2),
                                       dD3 = (D_3 - epi_D_3),
                                       dD3_r21 = (D_3 - epi_D_3_r21),
                                       dD3_l21 = (D_3 - epi_D_3_l21))

ggplot(data = vatd_indel_scores_25, aes(x=pos, y=dD2), na.rm=TRUE) + 
  scale_x_continuous(breaks=c(1, 50, 100, 150, 200)) +
  geom_hline(yintercept=0) +
  geom_line(aes(x=pos, y=dD2), color='red') +
  geom_line(aes(x=pos, y=dD3), color='blue') +
  labs(x = "Position", y = "Epistasis (Actual - single)") + 
  guides(fill="none") +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/vatd_25_single_epistastis.png", height = 5 , width = 10)

ggplot(data = vatd_indel_scores_25, aes(x=pos, y=dD2), na.rm=TRUE) + 
  scale_x_continuous(breaks=c(1, 50, 100, 150, 200)) +
  geom_hline(yintercept=0) +
  geom_line(aes(x=pos, y=dD3), color='blue') +
  geom_line(aes(x=pos, y=dD3_r21), color='darksalmon') +
  geom_line(aes(x=pos, y=dD3_l21), color='darkgoldenrod') +
  labs(x = "Position", y = "Epistasis (Actual - single)") + 
  guides(fill="none") +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/vatd_25_single_epistastis_3.png", height = 5 , width = 10)


ggplot(data = vatd_indel_scores_25) +
  scale_x_continuous(breaks=c(1, 50, 100, 150, 200)) +
  geom_line(aes(pos, epi_D_2), linetype=2) +
  geom_hline(yintercept=0) +
  geom_line(aes(pos, D_2), linetype=1) +
  geom_braid(aes(pos, ymin = epi_D_2, ymax = D_2, fill = epi_D_2 < D_2), alpha = 0.5) +
  guides(fill="none") +
  xlab("Position") + ylab("Surface score") +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/vatd_25_epi2_braid.png", height = 5 , width = 10)

ggplot(data = vatd_indel_scores_25) +
  scale_x_continuous(breaks=c(1, 50, 100, 150, 200)) +
  geom_line(aes(pos, epi_D_3), linetype=2) +
  geom_hline(yintercept=0) +
  geom_line(aes(pos, D_3), linetype=1) +
  geom_braid(aes(pos, ymin = epi_D_3, ymax = D_3, fill = epi_D_3 < D_3), alpha = 0.5) +
  guides(fill="none") +
  xlab("Position") + ylab("Surface score") +
  theme_classic()

ggsave("~/Desktop/dms_analysis/Plotting/Plots/vatd_25_epi3_braid.png", height = 5 , width = 10)

```